src/testapp/main.cpp

```diff
@@ -192,21 +180,22 @@ void debug_cli_loop()
         std::cout << "> ";
         std::cin >> command;
 
-        bool magnumOpen = ! g_testApp.m_renderer.m_sessions.empty();
-        if (auto const it = scenarios().find(command);
-            it != std::end(scenarios()))
+        bool const magnumOpen = ! g_testApp.m_renderer.m_sessions.empty();
+        auto const it = scenarios().find(command);
```

Comment:
in situations where the variable under test is not used outside the conditional block, it's preferred to use C++17's if-with-init to make the variable scoping as minimal as possible.

Reply:
I'd say for larger blocks of code it makes sense, but we disagree on this actually. To me, the perfect if statement only checks a boolean condition and nothing else. I mentioned it being "avoid if possible" in guidelines.md.